src/crimson/admin/osd_admin.cc

```diff
@@ -265,4 +272,96 @@ class SeastarMetricsHook : public AdminSocketHook {
 };
 template std::unique_ptr<AdminSocketHook> make_asok_hook<SeastarMetricsHook>();
 
+
+static ghobject_t test_ops_get_object_name(
+  const OSDMap& osdmap,
+  const cmdmap_t& cmdmap)
+{
+  int64_t pool;
+  if (auto poolarg = cmd_getval<std::string>(cmdmap, "pool"); poolarg) {
```

Comment:
might want to early return for error handling, like
```c++
auto pool = [&] {
  cmd_getval<std::string>(cmdmap, "pool");
  if (!pool_arg) {
    // throw
  }
  int64_t pool = osdmap.lookup_pg_pool_name(*poolarg);
  // more error handling
  return pool;
}();
```

Reply:
Not sure whether extra lambda improves readability. Anyway, I have no strong feelings here.

Reply:
i think the lambda is not the key to improve the readability here. the point is

- to reduce the indent level
- to highlight the happy path